tofu/utils.py

```diff
@@ -320,7 +320,15 @@ def save(obj, path=None, name=None, sep=None, deep=False, mode='npz',
     # Get stripped dictionnary
     deep = 'dict' if deep else 'ref'
     if sep is None:
-        sep = _SEP
+        if mode == 'mat':
```

Comment:
If a .mat file is being saved, the separator must be '_', otherwise a '.' would corrupt the mat file (minor change for robustness)